Migrate to Jetpack View Binding

alexmercerind opened the pull request:
[Jetpack View Binding](https://developer.android.com/topic/libraries/view-binding) is the recommended replacement for now deprecated [Butter Knife](https://github.com/JakeWharton/butterknife).

The migration away from [Butter Knife](https://github.com/JakeWharton/butterknife) is necessary in order to run the project on newer Android Studio & JDK versions. A [workaround](https://github.com/enviroCar/enviroCar-app/blob/dc9f95508b1db16b943d8b0a8ef8fdeaa9ea9dd5/gradle.properties#L14-L19) was introduced in #1000, in order to make the project run with [Butter Knife](https://github.com/JakeWharton/butterknife), which has now been removed as part of this pull-request.

SebaDro reviewed (changes requested):
There seems to be a little problem with the Logbook View. See inline comment.

In addition, in the [DashboardFragment ](https://github.com/enviroCar/enviroCar-app/blob/4f081ab1ccb73b64c5eff5a68c3d4ff74f903301/org.envirocar.app/src/org/envirocar/app/views/dashboard/DashboardFragment.java) the Bluetooth state button does not change automatically as Bluetooth state changes. E.g. the button should change it's color from red to blue automatically as soon as Bluetooth has been enabled on your device. Please, have a look on it.

alexmercerind commented:
Thank you very much for reviewing the pull-request. I have resolved the issues:

- https://github.com/enviroCar/enviroCar-app/pull/1002/commits/fe96a465d40c604dafdec9d832cc9a539e92f40f
  - It seems I accidentally caused an issue in this `ArrayAdapter`. I have corrected it & `convertView` is no longer `null`, similar to other `ArrayAdapter`s in the source-code .
- https://github.com/enviroCar/enviroCar-app/pull/1002/commits/f899e29a465c046b3a252d7fe68a4a35fe22c087
  - It seems I accidentally registered the wrong `BroadcastReceiver` in https://github.com/enviroCar/enviroCar-app/pull/1000/commits/40a203252ebf5583af8058b5510e8d0f9516a5b0 / #1000. This caused `BluetoothStateChangedEvent` not to be sent. I have corrected it.
  
 Thanks again!
